core/src/Streamly/Internal/Unicode/Parser.hs

```diff
@@ -294,5 +295,29 @@ signed p = (negate <$> (char '-' *> p)) <|> (char '+' *> p) <|> p
 -- \"Infinity\".
 --
 -- /Unimplemented/
-double :: Parser Char m Double
-double = undefined
+-- A strict coeff and exp pair
+data CE = CE !Integer {-# UNPACK #-}!Int
+
+double :: (Monad m) => Parser Char m Double
+double = do
```

Comment:
Can you write a benchmark for it? I suspect the performance will be poor because it is using Monad. We should write it directly by defining a step function.

Reply:
Also, enable the doctests in the documentation to check if they pass.